src/Microsoft.ML.ImageAnalytics/ImageLoader.cs

```diff
@@ -302,28 +300,42 @@ private static bool TryLoadDataIntoBuffer(string path, ref VBuffer<byte> imgData
 
                     count = (int)fileLength;
                     var editor = VBufferEditor.Create(ref imgData, count);
-                    bytesread = ReadToEnd(fs, editor.Values, readBuffer);
+                    bytesread = ReadToEnd(fs, editor.Values);
                     imgData = editor.Commit();
                     return (count > 0);
                 }
 
             }
 
-            private static int ReadToEnd(System.IO.Stream stream, Span<byte> bufferSpan, byte[] readBuffer)
+            private static int ReadToEnd(System.IO.Stream stream, Span<byte> bufferSpan)
             {
                 int totalBytesRead = 0;
                 int bytesRead;
-                var readBufferSpan = readBuffer.AsSpan();
-                var srcSpan = readBufferSpan;
-                while ((bytesRead = stream.Read(readBuffer, 0, readBuffer.Length)) > 0)
+
+                int chunksize = 4096; // Most optimal size for buffer, friendly to CPU's L1 cache
+                var bufferPool = ArrayPool<byte>.Shared;
+                byte[] readBuffer = bufferPool.Rent(chunksize);
```

Comment:
>Rent [](start = 47, length = 4)

I prefer the term "Lease" and have seen that in software stack but first time seeing "Rent"

Reply:
`Rent` and `Return` are the terms chosen in the `System.Buffers` library. They are used both for ArrayPool, and MemoryPool.

[https://docs.microsoft.com/en-us/dotnet/api/system.buffers.arraypool-1.rent](https://docs.microsoft.com/en-us/dotnet/api/system.buffers.arraypool-1.rent)
[https://docs.microsoft.com/en-us/dotnet/api/system.buffers.memorypool-1.rent](https://docs.microsoft.com/en-us/dotnet/api/system.buffers.memorypool-1.rent)

---
In reply to: [331301719](https://github.com/dotnet/machinelearning/pull/4293#discussion_r331301719) [](ancestors = 331301719)